src/openrct2/ride/ride.c

```diff
@@ -3450,153 +3490,157 @@ void ride_set_map_tooltip(rct_map_element *mapElement)
  * @param tuneId (bh)
  * @returns new position (ebp)
  */
-sint32 ride_music_params_update(sint16 x, sint16 y, sint16 z, uint8 rideIndex, uint16 sampleRate, uint32 position, uint8 *tuneId)
+sint32 ride_music_params_update(sint16 x, sint16 y, sint16 z, uint8 rideIndex, uint16 sampleRate, uint32 position, uint8 * tuneId)
 {
-    if (!(gScreenFlags & SCREEN_FLAGS_SCENARIO_EDITOR) && !gGameSoundsOff && g_music_tracking_viewport != NULL) {
-        LocationXY16 rotatedCoords = { 0, 0 };
+    if (!(gScreenFlags & SCREEN_FLAGS_SCENARIO_EDITOR) && !gGameSoundsOff && g_music_tracking_viewport != NULL)
+    {
+        LocationXY16 rotatedCoords = {0, 0};
 
-        switch (get_current_rotation()) {
-            case 0:
-                rotatedCoords.x = y - x;
-                rotatedCoords.y = ((y + x) / 2) - z;
-                break;
-            case 1:
-                rotatedCoords.x = -x - y;
-                rotatedCoords.y = ((y - x) / 2) - z;
-                break;
-            case 2:
-                rotatedCoords.x = x - y;
-                rotatedCoords.y = ((-y - x) / 2) - z;
-                break;
-            case 3:
-                rotatedCoords.x = y + x;
-                rotatedCoords.y = ((x - y) / 2) - z;
-                break;
+        switch (get_current_rotation())
+        {
+        case 0:
+            rotatedCoords.x = y - x;
+            rotatedCoords.y = ((y + x) / 2) - z;
+            break;
+        case 1:
+            rotatedCoords.x = -x - y;
+            rotatedCoords.y = ((y - x) / 2) - z;
+            break;
+        case 2:
+            rotatedCoords.x = x - y;
+            rotatedCoords.y = ((-y - x) / 2) - z;
+            break;
+        case 3:
+            rotatedCoords.x = y + x;
+            rotatedCoords.y = ((x - y) / 2) - z;
+            break;
         }
```

Comment:
The same switch block can be found in `window_ride_construction_draw_track_piece`, so you could extract this to a new function in map for both usages.

Reply:
It's not entirely the same kind of calculation, by the look of it.

Reply:
https://github.com/OpenRCT2/OpenRCT2/blob/5c98c7ee5eefacffd4df03f57680955021fb608b/src/openrct2/windows/RideConstruction.cpp#L2286-2304

Reply:
Ok, I split it off.